### PR TITLE
creating a data model

### PR DESCRIPTION
**Pull request by Zeph180**
Build your project schema.
Create and run the necessary migration files.
Table and column names should match the ERD diagram.
note: photo for users table should be a link to an image
Foreign keys should be included.
Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Zeph180,

Good job so far!
✔️ Use of PostgreSQL for database
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
I am requesting these changes here because you have not changed the related files.

- [ ] You are missing the table attributes for the models. You have only posts and users' tables. Kindly also ensure you use the correct table columns and types.

  image

  From the image above
  
  - Your posts table should have `author_id`, not `user_id`. What you should have is `text`, not `body`. You are also missing the `likes_counter` and `comments_counter` (both integer types) table columns
  - Your users' table should have `name` (string), `photo` (string), `posts_counter` (integer), `bio` (text). Kindly remove `email`, `password_digest` and `photo_url` as you current have.

- [ ] You are missing the **comments** and **likes** tables, and do not have migrations for them. Kindly generate migrations for the comments and likes and ensure they stick to the table column names required in the guidelines.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — COMMENTED** [review]
# Hi @Zeph180 😔, 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that dies not have a descriptive README file. Kindly use this [Template ](https://github.com/microverseinc/readme-template#readme) and edit according to your project to ensure you have a professional and descriptive README file 🙂 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: db/schema.rb [hunk 1]
```diff
@@ -10,26 +10,49 @@
...
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
+
+  create_table "likes", force: :cascade do |t|
+    t.bigint "user_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
```
Comment (Davidosky007) [inline db/schema.rb]: * As shown in the ERD diagram you should have a foreign key that will create author_id and a corresponding index for both comments and likes
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Zeph180 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Good looking readme 👍 
* Proper use of Gitflow 👍 
* Migrations used to maintain database 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @Zeph180 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

weldone


## Highlights
:heavy_check_mark: Added foreign keys to comment and likes
:heavy_check_mark: Great effort so far


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**